compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -131,7 +131,9 @@ def __init__(
         }
         log.info(f"Platform info: {self.current_platform}")
 
-        self.results_passthrough: queue.Queue[_ResultPassthroughType] = queue.Queue()
+        self.results_passthrough: queue.Queue[_ResultPassthroughType | None] = (
+            queue.Queue()
```

Comment:
You do note that None is a poison pill several times below, but leaving a comment here instead (or in addition to) might be more visible when reading the code to understand its usage/path IMO.